src/types/near_allowance.rs

```diff
@@ -0,0 +1,91 @@
+const UNLIMITED: &str = "unlimited";
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    serde::Serialize,
+    serde::Deserialize,
+    derive_more::AsRef,
+    derive_more::From,
+    derive_more::Into,
+)]
+#[as_ref(forward)]
+pub struct NearAllowance(Option<crate::types::near_token::NearToken>);
+
+impl std::fmt::Display for NearAllowance {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(amount) = self.0 {
+            amount.fmt(f)
+        } else {
+            write!(f, "{UNLIMITED}")
+        }
+    }
+}
+
+impl std::str::FromStr for NearAllowance {
+    type Err = color_eyre::eyre::Report;
```

Comment:
Let's try using `near_token::NearTokenError` for consistency

```suggestion
    type Err = near_token::NearTokenError;
```